### ISSUE TITLE
ArrayBuilder is not Send

### ISSUE THREAD

**Issue by gz** [chmp/serde_arrow#225]
I'm getting

```
190 |             let mut insert_builder = Arc::new(ArrayBuilder::new(sas).expect("can build array"));
    |                 ------------------ has type `Arc<serde_arrow::ArrayBuilder>` which is not `Send`

```

if I try to use ArrayBuilder in async functions (which often have the Send constraint for the things in the closure). I think the problem is the raw pointer here:

https://github.com/chmp/serde_arrow/blob/eb8d37a5bdab748251aa983cb1c1517047f28702/serde_arrow/src/internal/serialization/struct_builder.rs#L23C1-L23C55

Maybe it could be rewritten to not use raw pointers. Or be declared as Send if the raw pointer isn't used in a way that breaks Send guarantees (from a quick glance I don't think it is).

**Comment by chmp** [comment]
@gz  Thanks for the report. You're right there is no reason for the structure to not be Send. I will fix this issue on the develop-0.12 branch.

**Comment by chmp** [comment]
Fixed with #227. I will close this issue once I release `0.12`. The refactoring of the error messages will take some time though

**Comment by gz** [comment]
Awesome thanks for the quick turnaround!

**Comment by chmp** [comment]
Released with with 0.12.0
